Hotel 16

dudarenko-dmitry opened the pull request:
need help in:
8. Build the application in WAR and deploy it on Tomcat.

I'm not able to solve issue with error:
![image](https://github.com/dudarenko-dmitry/senla_training/assets/102149674/06355b26-5379-4af8-886a-0adf52dcc726)

dudarenko-dmitry committed:
Task_16:
1. Create new branch.
2. Edit Readme

dudarenko-dmitry committed:
Task_16:
1. Create new branch.
2. Edit Readme.
3. Remove module UI.
4. update pom.
5. create DTO and Mappers for all entities.
6. Updates all Controllers' methods for using REST.
7. Refactored some CRUD methods.
8. add web.xml
9. change to WAR in pom

dudarenko-dmitry committed:
Task_16:
1. Remove module UI.

dudarenko-dmitry committed:
Task_16:
1. update pom

dudarenko-dmitry committed:
Task_16:
1. create DTO for all Entities.
2. refactored Controller's and Service's interfaces.
3. refactored Guest's Controller, Service and Dao layers.
4. refactored Room's Controller, Service and Dao layers.

dudarenko-dmitry committed:
Task_16:
1. create DTORead and mapper for Order Entity.
2. refactored Order's Controller and Service layers.

dudarenko-dmitry committed:
Task_16:
1. create DTORead and mapper for HotelService Entity.
2. refactored HotelService's Controller and Service layers.

dudarenko-dmitry committed:
Task_16:
1. add web.xml

dudarenko-dmitry committed:
Task_16:
Create all REST methods

dudarenko-dmitry committed:
Task_16:
Create all REST methods

dudarenko-dmitry committed:
Task_16:
change to WAR in pom

dudarenko-dmitry committed:
Merge remote-tracking branch 'origin/Hotel_16' into Hotel_16

dudarenko-dmitry committed:
Task_16:
add @Transactional

Mark005 commented on src/main/java/pl/senla/hotel/controller/ControllerFacilitySpring.java:
```diff
@@ -0,0 +1,122 @@
+    }
+
+    @Override
+    @GetMapping("/sort=category")
```
better to provide it as query params http://myapp.com/books?sort=author like here and use only one controller method for all sorting

dudarenko-dmitry replied:
I can't understand why query params doesn't work in my app...

Mark005 commented on src/main/java/pl/senla/hotel/controller/ControllerOrderSpring.java:
```diff
@@ -0,0 +1,75 @@
+    public List<OrderReadDto> readAll() {
+        log.debug("ControllerOrder call ServiceOrder's method 'readAll'.");
+        return OrderReadDtoMapperUtil
+                .convertListOrderToOrderDtoRead(orderService.readAll(), serviceRoomReservation.readAll());
```
you don't need to do this because each order will have all the attached services inside

dudarenko-dmitry replied:
ok. ready

Mark005 commented on src/main/java/pl/senla/hotel/controller/ControllerOrderSpring.java:
```diff
@@ -0,0 +1,75 @@
+            InvocationTargetException, NoSuchMethodException, InstantiationException {
+        log.debug("ControllerOrder call ServiceOrder's method 'create'.");
+        Order order = orderService.create(orderCreateDto);
+        return OrderReadDtoMapperUtil.convertOrderToOrderDtoRead(order, new ArrayList<>());
```
same in other methods, so you can create two different methods to map to flat dto and map with services

dudarenko-dmitry replied:
ready

Mark005 commented on src/main/java/pl/senla/hotel/controller/ControllerFacility.java:
```diff
@@ -0,0 +1,22 @@
+
+public interface ControllerFacility extends ControllerCRUDALL<RoomDto, RoomDto> {
+
+    RoomDto updateRoomStatusAvailable(Integer idRoom) throws InvocationTargetException,
```
why do your controllers throw exceptions?? if you have a problem with checked exceptions then solve it in the service, otherwise if it's unchecked add controller advice to catch them and convert to the same error response

Mark005 commented on src/main/java/pl/senla/hotel/dto/GuestDto.java:
```diff
@@ -0,0 +1,19 @@
+public class GuestDto implements Serializable {
+
+    @Serial
+    private static final long serialVersionUID = 112L;
```
why do you need this?

Mark005 commented on src/main/java/pl/senla/hotel/dto/GuestDto.java:
```diff
@@ -0,0 +1,19 @@
+    private static final long serialVersionUID = 112L;
+
+    private String name;
+    private Integer phoneNumber;
```
string mb?

Mark005 commented on src/main/java/pl/senla/hotel/dto/HotelServiceCreateDto.java:
```diff
@@ -0,0 +1,32 @@
+    private Integer idGuest;
+    private Integer idHotelFacility;
+    private Integer numberOfDays;
+    private String checkInTimeString;
```
you can provide it as a Local/Zoned dateTime or just Date depends on what you used in entity

Mark005 commented on src/main/java/pl/senla/hotel/dto/OrderReadDto.java:
```diff
@@ -0,0 +1,29 @@
+    private List<HotelServiceReadDto> hotelServices;
+
+    @Override
+    public String toString() {
```
if you are using the Lombok you can read what @Data doing

Mark005 commented on src/main/java/pl/senla/hotel/exceptions/GuestNotFoundAdvise.java:
```diff
@@ -0,0 +1,18 @@
+    @ExceptionHandler(GuestNotFoundException.class)
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+    String guestNotFoundHandler(GuestNotFoundException exception) {
+        return exception.getMessage();
```
use DTO to provide response as JSON and not as a String

Mark005 commented on src/main/java/pl/senla/hotel/exceptions/HotelServiceNotFoundAdvise.java:
```diff
@@ -0,0 +1,18 @@
+@ControllerAdvice
+public class HotelServiceNotFoundAdvise {
+
+    @ResponseBody
```
move all in one controller advice

Mark005 commented on src/main/java/pl/senla/hotel/utils/HotelServiceReadDtoMapperUtil.java:
```diff
@@ -0,0 +1,109 @@
+                .toList();
+    }
+
+    public static LocalDateTime convertStringToTime(String checkedDateString) {
```
spring will do it automatically, just put the necessary type in consuming DTO, but you will have to provide it in the standard form

dudarenko-dmitry committed:
Task_16:
refactored ReadAll in ControllerFacilitySpring

dudarenko-dmitry committed:
Task_16:
refactored ReadAll in ControllerFacilitySpring

dudarenko-dmitry committed:
Task_16:
refactored ReadAll in ControllerGuestSpring

dudarenko-dmitry committed:
Task_16:
1. refactored ReadAll in ControllerOrderSpring.
2. Refactored OrderReadDtoMapperUtil

dudarenko-dmitry committed:
Task_16:
1. refactored ReadAll in ControllerOrderSpring.
2. Refactored OrderReadDtoMapperUtil
